evap/student/templates/student_index_semester_evaluations_list.html

```diff
@@ -0,0 +1,164 @@
+                                                {% if evaluation.is_single_result %}
+                                                    ({{ evaluation.vote_start_datetime|date }})
+                                                {% endif %}
+                                            </span>
+                                            {% if can_download_grades and not semester.grade_documents_are_deleted %}
+                                                {% if course.evaluation_count == 1 %}
+                                                    {% if course.grade_documents.count > 1 %}
+                                                        <div class="btn-group" role="group">
+                                                    {% endif %}
+                                                    {% for grade_document in course.grade_documents.all %}
+                                                        <a class="btn btn-sm btn-sm-narrow btn-light btn-grade-download" href="{% url 'grades:download_grades' grade_document.id %}" role="button"
+                                                            data-toggle="tooltip" data-placement="top" title="{{ grade_document.description }}">
+                                                            <span class="far fa-file"></span>
+                                                        </a>
+                                                    {% endfor %}
+                                                    {% if course.grade_documents.count > 1 %}
+                                                        </div>
+                                                    {% endif %}
+                                                {% endif %}
+                                            {% endif %}
+                                        </div>
+                                        {% if course.evaluation_count == 1 %}
+                                            <span class="badge badge-secondary">{{ course.type }}</span>
+                                        {% endif %}
+                                        {% if evaluation.is_midterm_evaluation %}<span class="badge badge-dark">{% trans 'Midterm evaluation' %}</span>{% endif %}
+                                        {% if evaluation.is_single_result %}
+                                            <span class="badge badge-success">{% trans 'Single result' %}</span>
+                                        {% endif %}
+                                        {% if course.evaluation_count == 1 %}
+                                            <i class="small">{{ course.responsibles_names }}</i>
+                                        {% endif %}
+                                    </td>
+                                    <td style="width: 20%">
+                                        {% if semester.results_are_archived %}
+                                        {% elif evaluation.is_single_result %}
```

Comment:
huh?

Reply:
in case the results are archived nothing should be shown there. the cases could be written in a different order with additional checks for not archived instead :)

Reply:
Ah I see, so its basically a guard clause, but you can't return or do any other kind of early exit, so you just put an empty body. It kind of makes sense. If rewriting makes the code more complex, we can also leave it like this.